### ISSUE TITLE
What is the correct method to utilize UDatatable using UnLua?

### ISSUE THREAD

**Issue by futouyiba** [Tencent/UnLua#116]
Data tables are essential in game dev, and UDatatable is the recommended way to dev in a data-driven style. But if we are using UnLua, how to get stuff done? For example:

![image](https://user-images.githubusercontent.com/6793989/75432570-8ed5f680-5989-11ea-8e31-efa227bfba2a.png)

but if we want to convert it to Lua:
![image](https://user-images.githubusercontent.com/6793989/75432650-b200a600-5989-11ea-8948-7e82c2353a36.png)
it will pop errors, because row doesn't have the property(cost).

So basically UDatatable is generic for all struct types. But it could not casted to specific type in Lua, could it?

![image](https://user-images.githubusercontent.com/6793989/75432351-3ef72f80-5989-11ea-8faa-8b05f461c0ce.png)

**Comment by rowechien** [comment]
UnLua can handle this case easily. It's a piece of cake... and I think you need some codes  (*^_^*) 
row struct:
![image](https://user-images.githubusercontent.com/7972243/75622943-917e5900-5be0-11ea-9459-38403fd34445.png)
data table:
![image](https://user-images.githubusercontent.com/7972243/75622977-cdb1b980-5be0-11ea-8d61-0623df11f3d0.png)
lua codes:
![image](https://user-images.githubusercontent.com/7972243/75623016-05b8fc80-5be1-11ea-9253-8b289db1b0e6.png)
and the results:
![image](https://user-images.githubusercontent.com/7972243/75623026-1a959000-5be1-11ea-8237-33a7de25ac96.png)

**Comment by lvpengfeifei** [comment]
but if the struct be defined by blueprint. will it work??

**Comment by hugebug4ever** [comment]
> but if the struct be defined by blueprint. will it work??

"struct defined by bp " also works as expected.
PS. You have to refrene the struct defined in bp with "F" prefix manually

**Comment by hugebug4ever** [comment]
BP structure worked for 4.26 & unlua 1.0, while not 4.27 & unlua 2.0.
Has not figured out how to bypass it yet.

**Comment by lvpengfeifei** [comment]
邮件已经收到啦
